src/site/topic24.rst

```diff
@@ -174,6 +174,96 @@ Minimum & Maximum
 Remove Minimum & Maximum
 ========================
 
+* Finding the minimum and maximum values in the linked binary search tree is relatively simple
+* However, *removing* from the tree adds extra complexity as we need to preserve the binary search tree ordering of our tree
+
+**Remove Minimum**
+
+.. code-block:: java
+    :linenos:
+
+    public T removeMin() {
+        T returnElement = null;
+        if (isEmpty()) {
+            throw new NoSuchElementException();
+        } else if (root.getLeft() == null) {
+            returnElement = root.getData();
+            root = root.getRight();
+        } else {
+            returnElement = removeMin(root, root.getLeft());
+        }
+        size--;
+        return returnElement;
+    }
+
+    private T removeMin(Node<T> current, Node<T> leftChild) {
+        if (leftChild.getLeft() == null) {
+            current.setLeft(leftChild.getRight());
+            return leftChild.getData();
+        } else {
+            return removeMin(current.getLeft(), leftChild.getLeft());
+        }
+    }
+
+* Above are two functions that work together to remove the minimum value
+
+* First, take note of the public method
+    * It checks if the tree is empty
+    * It checks if the root happens to be the left most node
+        * If it is, then make the root's right child the new root of the whole tree
+    * Otherwise, call the private recursive method looking down the left subtree
+
+* If the public method doesn't remove the minimum, the recursive private method is the one that goes looking for the minimum to remove
+* At this point we know that ``leftChild`` exists (is not ``null``) since this was checked in the calling method
+    * Remember, if the root's left child does not exist, then the root must contain the minimum value in the tree
+
+* We'll start with the recursive case --- if ``leftChild``'s left child does exist, then we call the recursive function again to keep looking down the left subtrees
+
+* If ``LeftChild``'s left subtree does **not** exist, then we know that ``leftChild`` contains the minimum value
+    * Simply replace ``current``'s left child (which ``leftChild`` also references) with ``leftChild``'s right child
+        * It does not matter if the left child's right child is ``null`` or not, it works either way
+
+
+**Remove Maximum**
+
+* We could implement the ``removeMax`` with the same idea, but to show an alternative idea, an iterative method will be used instead
```

Comment:
I saw the code block and was like "uhhh..." then read it'd be iterative. Good way to compare/contrast the two approaches.